symbulate/results.py

```diff
@@ -330,61 +364,205 @@ def plot(self, type=None, alpha=None, normalize=True, jitter=False,
                 if jitter:
                     x += np.random.normal(loc=0, scale=.01 * (max(x) - min(x)), size=len(x))
                     y += np.random.normal(loc=0, scale=.01 * (max(y) - min(y)), size=len(y))
-                # get next color in cycle
-                axes = plt.gca()
-                color = get_next_color(axes)
-                plt.scatter(x, y, color=color, alpha=alpha, **kwargs)
-            else:
-                raise Exception("I don't know how to plot these variables.")
+                ax.scatter(x, y, alpha=alpha, c=color, **kwargs)
+            elif 'hist' in type:
+                histo = ax.hist2d(x, y, bins=bins, cmap='Blues')
+                if 'marginal' not in type:
+                    caxes = fig.add_axes([0, 0.1, 0.05, 0.8])
+                else:
+                    caxes = fig.add_axes([0, 0.1, 0.05, 0.57])
+                cbar = plt.colorbar(mappable=histo[3], cax=caxes)
+                caxes.yaxis.set_ticks_position('left')
+                cbar.set_label('Density')
+                caxes.yaxis.set_label_position("left")
+                new_labels = []
+                for label in caxes.get_yticklabels():
+                    new_labels.append(int(label.get_text()) / len(x))
+                caxes.set_yticklabels(new_labels)
+            elif 'density' in type:
+                res = np.vstack([x, y])
+                density = gaussian_kde(res)
+                xmax = max(x)
+                xmin = min(x)
+                ymax = max(y)
+                ymin = min(y)
+                Xgrid, Ygrid = np.meshgrid(np.linspace(xmin, xmax, 100),
+                                           np.linspace(ymin, ymax, 100))
+                Z = density.evaluate(np.vstack([Xgrid.ravel(), Ygrid.ravel()]))
+                den = ax.imshow(Z.reshape(Xgrid.shape), origin='lower', cmap='Blues',
+                          aspect='auto', extent=[xmin, xmax, ymin, ymax]
+                )
+                if 'marginal' not in type:
+                    caxes = fig.add_axes([0, 0.1, 0.05, 0.8])
+                else:
+                    caxes = fig.add_axes([0, 0.1, 0.05, 0.57])
+                cbar = plt.colorbar(mappable=den, cax=caxes)
+                caxes.yaxis.set_ticks_position('left')
+                cbar.set_label('Density')
+                caxes.yaxis.set_label_position("left")
+            elif 'tile' in type:
+                counts = self._get_counts()
+                x_uniq = np.unique(x)
+                y_uniq = np.unique(y)
+                xmin = min(x)
+                ymin = min(y)
+                xmax = max(x)
+                ymax = max(y)
+                x_pos = np.arange((xmax + 1) - xmin)
+                y_pos = np.arange((ymax + 1) - ymin)
+                nums = len(x)
+
+                intensity = np.zeros(shape=(len(y_pos), len(x_pos)))
+                for key, val in counts.items():
+                    intensity[key[1] - ymin][key[0] - xmin] = val / nums
```

Comment:
This code won't work if x and y contain non-integer values. You should really use the values of x_pos and y_pos, since those are guaranteed to be 0 to (xmax - xmin).

Reply:
In the 2 discrete variable case, won't all the values be integers though? If 'tile' is passed into plot when the variables are continuous, there is a check that will change the type to scatter.

Reply:
Not necessarily. A discrete random variable could take on the values 0, 0.5, 1.0, 1.5, ...